### PR TITLE
Feature/8 filter by application state

### PR DESCRIPTION
**Pull request by clriesco**
- Added state to card info
- Added filter by state

### CODE REVIEW

**Comment by clriesco** [comment]
#7 
#8

File: src/app/page.tsx [hunk 1]
```diff
@@ -14,12 +14,34 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
...
+
+  const fuse = new Fuse(filteredData, {
+    keys: [
+      'info.core_information.data_owner_name',
+      'info.core_information.data_owner_region',
+      'info.core_information.data_owner_industry',
+    ],
+  })
+  const results = fuse.search(searchTerm)
+  console.log({ results })
```
Comment (jbesraa) [inline src/app/page.tsx]: remove log please

File: src/app/page.tsx [hunk 2]
```diff
@@ -14,12 +14,34 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
...
+  const [filter, setFilter] = useState('all')
+  const [searchTerm, setSearchTerm] = useState('')
+  const filteredData =
+    data?.filter(
+      (app) =>
+        filter === 'all' || app.info.application_lifecycle.state === filter,
+    ) || []
+
+  const fuse = new Fuse(filteredData, {
+    keys: [
```
Comment (jbesraa) [inline src/app/page.tsx]: I would get the keys from one object and put them here, for example: `Object.keys(data[0])`
you need to make sure data[0] is exist though

Comment (jbesraa) [reply src/app/page.tsx]: I dont see also where fuse is being triggered. 
maybe add `useEffect` that only runs when `searchTerm` is changed. also when you do search/filter you need to take into account the other parameter, ie:
when there is search, u need to also if any filters are set and when a filter is changed u need to also check if there is a search term

File: src/app/page.tsx [hunk 3]
```diff
@@ -57,11 +85,13 @@ export default function Home() {
 
         <TabsContent value="grid">
           <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6 ">
-            {data?.map((app) => <AppCard application={app} key={app.id} />)}
+            {searchResults?.map((app) => (
```
Comment (jbesraa) [inline src/app/page.tsx]: Lets add type to `app` here

**Review by jbesraa — CHANGES_REQUESTED** [review]
looks good! added a couple of comments

File: src/app/page.tsx [hunk 4]
```diff
@@ -14,12 +14,43 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
 import { getAllApplications } from '@/lib/apiClient'
 import { Search } from 'lucide-react'
 import { useQuery } from 'react-query'
+import { useEffect, useState } from 'react'
+import Fuse from 'fuse.js'
+import { Application } from '@/type'
```
Comment (jbesraa) [inline src/app/page.tsx:19]: @clriesco where is this defined?

Comment (jbesraa) [reply src/app/page.tsx:19]: the `Application` type

**Review by jbesraa — APPROVED** [review]

**Pull request merged**